pkg/volume/validation/pv_validation.go

```diff
@@ -69,3 +70,13 @@ func ValidatePathNoBacksteps(targetPath string) error {
 
 	return nil
 }
+
+// ValidateVolumeIsConfigMap will check volume is a configMap
+func ValidateVolumeIsConfigMap(volumes []v1.Volume, mountName string) bool {
+	for _, volume := range volumes {
+		if volume.Name == mountName && volume.ConfigMap != nil {
```

Comment:
I am a little confused, why do we only check configmap?
I noticed that there are many resources in volume, do they also need to be considered?

Reply:
ConfigMap is the only source that can be treated as a "File". If the Volume source is a file system or a block, we can't check whether subPath exists or not before the pod starts.

Reply:
Got it. Thanks

Reply:
secret, downwardapi should also work